### PR TITLE
Run compiler build for profiling in temporary directory

### PR DESCRIPTION
**Pull request by milan-tom**
If the compiler build fails during profiling or the script is terminated during the build, this leaves the build in the repository being developed on in a bad state. This PR fixes this problem by cloning the repository to a temporary directory (another directory within the parent of the directory that the local repository is stored in i.e. a sibling) and then running the build in that temporary directory leaving the original unchanged (aside from changes in the profile file dump directory and the summary CSV being added).

### CODE REVIEW

File: scripts/profile-compiler-build.sh [hunk 1]
```diff
@@ -1,11 +1,13 @@
 #!/bin/sh
 
-# To be run from the root of the Flambda backend repo
-
 set -e -u -o pipefail
 
-dump_dir="`pwd`/_profile"
-summary_path="`pwd`/summary.csv"
+# Works regardless of where script run from
+scripts_dir=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
```
Comment (xclerc) [inline scripts/profile-compiler-build.sh:6]:
```suggestion
scripts_dir=$( cd "$( dirname "$0" )" &> /dev/null && pwd )
```
Is that version enough to achieve what we want?
(I am just worried the current version is less
compatible with various shells.)

File: scripts/profile-compiler-build.sh [hunk 2]
```diff
@@ -1,11 +1,13 @@
...
+
+root=$scripts_dir/..
```
Comment (xclerc) [inline scripts/profile-compiler-build.sh:8]:
```suggestion
root="$scripts_dir/.."
```

File: scripts/profile-compiler-build.sh [hunk 3]
```diff
@@ -37,7 +39,14 @@ build_compiler() {
   make install
 }
 
+temp_dir=$(mktemp -d -p $root/.. -t tmp.XXXXXXX)
```
Comment (xclerc) [inline scripts/profile-compiler-build.sh:42]: Not sure about the `-p $root/..` parameter;
is relying on `TMPDIR` a problem in practice?

**Review by xclerc — COMMENTED** [review]
Looks good; only a couple of minor
remarks/questions. I think it is a good
property to not longer depend on
the directory from which the script
is launched.

File: scripts/profile-compiler-build.sh (hunk 1 above)
Comment (milan-tom) [reply scripts/profile-compiler-build.sh:6]: Yes, this is sufficient.

File: scripts/profile-compiler-build.sh (hunk 3 above)
Comment (milan-tom) [reply scripts/profile-compiler-build.sh]: As discussed offline, `TMPDIR` is not always set and may have space issues. So, the code has been changed to make the temp directory in `TMPDIR` if it is set and as a sibling of the repository directory if it is not set.

**Review by xclerc — APPROVED** [review]

**Pull request merged**